pkg/controller/manager.go

```diff
@@ -94,7 +112,7 @@ func (m *Manager) createControllerLocked(name string, params ControllerParams) *
 			group:      params.Group,
 			uuid:       uuid.New().String(),
 			stop:       make(chan struct{}),
-			update:     make(chan ControllerParams, 1),
+			update:     make(chan ControllerParams, 32),
```

Comment:
Why does this channel capacity need to change?

The concern I have is that we *must* make this code robust to blocking sends on the channel. Channel size is primarily useful as a buffer, not to handle concurrency issues. I don't think we expect to see many controller params updates, so maybe this should just stay at capacity 1?

Reply:
Ok, I change it to be 1.